### PR TITLE
[BEAM-1992] Move Count.PerElement javadoc to a public place

### PR DESCRIPTION
**Pull request by wtanaka**
- [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by wtanaka** [comment]
@jkff

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Count.java [hunk 1]
```diff
@@ -101,6 +89,18 @@ private Count() {
    * PCollection<KV<String, Long>> wordCounts =
    *     words.apply(Count.<String>perElement());
    * } </pre>
+   */
+  public static <T> PTransform<PCollection<T>, PCollection<KV<T, Long>>> perElement() {
+    return new PerElement<>();
+  }
+
+  /**
+   * {@code Count.PerElement<T>} takes a {@code PCollection<T>} and returns a
```
Comment (jkff) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Count.java]: It seems reasonable to keep this block in the javadoc of perElement(), and reduce the javadoc of PerElement to simply `Implementation of {@link #perElement}.`

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Count.java [hunk 2]
```diff
@@ -74,21 +74,9 @@ private Count() {
    * Returns a {@link PerElement Count.PerElement} {@link PTransform} that counts the number of
```
Comment (jkff) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Count.java]: Mind also cleaning up this reference? It now refers to a private class and is useless.

Comment (wtanaka) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Count.java]: replacing with
```
   * Returns a {@link PTransform} that counts the number of occurrences of each element
   * in its input {@link PCollection}.
```

**Pull request closed without merging**